test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -846,5 +848,59 @@ public void TensorFlowTransformCifarInvalidShape()
+            var dataView = mlContext.Data.ReadFromEnumerable(data);
+
+            var lookupMap = mlContext.Data.ReadFromTextFile(@"sentiment_model/imdb_word_index.csv",
+                   columns: new[]
+                   {
+                        new TextLoader.Column("Words", DataKind.TX, 0),
+                        new TextLoader.Column("Ids", DataKind.I4, 1),
+                   },
+                   separatorChar: ','
+               );
+
+            // We cannot resize variable length vector to fixed length vector in ML.Net
+            // The trick here is to create two pipelines.
+            // The first pipeline tokenzies the strings into words and maps the words to an integer which is an index in the dictionary.
+            // Then this integer vector is retrieved from the pipeline and resized to fixed length.
+            // The second pipeline takes the resized integer vector and passed to TensoFlow and get the classification scores.
+            var estimator = mlContext.Transforms.Text.TokenizeWords("Sentiment_Text", "TokenizedWords")
+                .Append(mlContext.Transforms.Conversion.ValueMap(lookupMap, "Words", "Ids", new[] { ("TokenizedWords", "Features") }));
+            var dataPipe = estimator.Fit(dataView)
+                .CreatePredictionEngine<TensorFlowSentiment, TensorFlowSentiment>(mlContext);
+
+            // For explanation on how was the `sentiment_model` created 
+            // c.f. https://github.com/dotnet/machinelearning-testdata/blob/master/Microsoft.ML.TensorFlow.TestModels/sentiment_model/README.md
+            string modelLocation = @"sentiment_model";
+            var tfEnginePipe = mlContext.Transforms.ScoreTensorFlowModel(modelLocation, new[] { "Features" }, new[] { "Prediction/Softmax" })
+                .Append(mlContext.Transforms.CopyColumns(("Prediction/Softmax", "Prediction")))
+                .Fit(dataView)
+                .CreatePredictionEngine<TensorFlowSentiment, TensorFlowSentiment>(mlContext);
+
+            var processedData = dataPipe.Predict(data[0]);
+            Array.Resize(ref processedData.Features, 600);
+            var prediction = tfEnginePipe.Predict(processedData);
+
+            Assert.Equal(2, prediction.Prediction.Length);
+            Assert.InRange(prediction.Prediction[1], 0.650032759 - 0.01, 0.650032759 + 0.01);
```

Comment:
Please use Assert.Equal. If there are only two prediction values, can we check them all?

Reply:
What do you mean by Assert.Equal? Here we are checking the range within particular threshold. 
No need to check another value. These are probabilities.

---
In reply to: [252005383](https://github.com/dotnet/machinelearning/pull/2302#discussion_r252005383) [](ancestors = 252005383)

Reply:
ok, I got you what you meant with Assert.Equal. I actually want to check if my values are in range e.g. 0.64 <= prediction <= 0.66 which I cannot do with Assert.Equal, can I?
Also, I feel `InRange` more readable than other when asserting thresholds.

---
In reply to: [252006645](https://github.com/dotnet/machinelearning/pull/2302#discussion_r252006645) [](ancestors = 252006645,252005383)

Reply:
You have tolerance in Assert.Equal.

Reply:
It uses number of decimal places which is not applicable here.

---
In reply to: [252050957](https://github.com/dotnet/machinelearning/pull/2302#discussion_r252050957) [](ancestors = 252050957)

Reply:
We could trim 0.650032759 to 0.65.